chains-examples/docs/whisper-vad/chunker/vad.py

```diff
@@ -0,0 +1,130 @@
+import base64
+import functools
+import io
+import time
+from typing import AsyncIterator, List, NamedTuple, Optional
+
+import torch
+import utilities.data_types as data_types
+from silero_vad import load_cpu_model
+from silero_vad.utils_vad import BatchSettings, get_speech_timestamps
+from utilities.helpers import AudioStreamer
+
+
+@functools.lru_cache
+def get_vad_model():
+    # TODO: check gpu availability, if gpu is not aviailable just use the normal load model
+    """Returns the VAD model instance."""
+    return load_cpu_model()
+
+
+class VadOptions(NamedTuple):
+    """VAD options.
+
+    Attributes:
+      threshold: Speech threshold. Silero VAD outputs speech probabilities for each audio chunk,
+        probabilities ABOVE this value are considered as SPEECH. It is better to tune this
+        parameter for each dataset separately, but "lazy" 0.5 is pretty good for most datasets.
+      min_speech_duration_ms: Final speech chunks shorter min_speech_duration_ms are thrown out.
+      max_speech_duration_s: Maximum duration of speech chunks in seconds. Chunks longer
+        than max_speech_duration_s will be split at the timestamp of the last silence that
+        lasts more than 100ms (if any), to prevent aggressive cutting. Otherwise, they will be
+        split aggressively just before max_speech_duration_s.
+      min_silence_duration_ms: In the end of each speech chunk wait for min_silence_duration_ms
+        before separating it
+      speech_pad_ms: Final speech chunks are padded by speech_pad_ms each side
+    """
+
+    threshold: float = 0.5
+    min_speech_duration_ms: int = 250
+    max_speech_duration_s: float = float("inf")
+    min_silence_duration_ms: int = 2000
+    speech_pad_ms: int = 400
+
```

Comment:
Would be good to move this to types and allow it to be configurable with the audio input